Rename Analytics rules to Smart rules (New) (#3714)
SoftwareAG/c8y-docs: content/streaming-analytics/smart-rules-plugin-bundle/getting-started.md

---
weight: 10

> title: Getting started with the Smart rules 2.0 plugin

Reviewer: ```suggestion
title: Getting started with the Smart rules (NEW) plugin
```

layout: redirect
---

> ### What is the Smart rules plugin {#what-is-the-smart-rules-plugin}

Reviewer 2: ```suggestion
### What is the Smart rules (NEW) plugin {#what-is-the-smart-rules-plugin}
```

> ### What is the Smart rules 2.0 plugin {#what-is-the-smart-rules-2.0-plugin}

Reviewer: ```suggestion
### What is the Smart rules plugin {#what-is-the-smart-rules-plugin}
```

We can just use "Smart rules" here, the top-level title is enough to show we're talking about the "new" one

> The Smart rules plugin extends the Analytics Builder capabilities within Streaming Analytics by enabling users to create and manage Analytics Builder model instances directly from device and group contexts within applications such as Device Management and Cockpit.

Reviewer: and Digital Twin Manager

> The Smart rules 2.0 plugin extends the Analytics Builder capabilities within Streaming Analytics by enabling users to create and manage Analytics Builder model instances directly from device and group contexts within applications such as Device Management and Cockpit.

Reviewer: ```suggestion
The Smart rules plugin extends the Analytics Builder capabilities within Streaming Analytics by enabling users to create and manage Analytics Builder model instances directly from device and group contexts within applications such as Device Management and Cockpit.
```

> The Smart rules plugin extends the Analytics Builder capabilities within Streaming Analytics by enabling users to create and manage Analytics Builder model instances directly from device and group contexts within applications such as Device Management, Cockpit and Digital Twin Manager.

Reviewer 2: ```suggestion
The Smart rules (NEW) plugin extends the Analytics Builder capabilities within Streaming Analytics by enabling users to create and manage Analytics Builder model instances directly from device and group contexts within applications such as Device Management, Cockpit, and Digital Twin Manager.
```

Reviewer: We don't want to do this kind of change as we're already in the contetx of new smart rules - we have a PM decision not to keep repeating NEW all over the documentation text in places where the context is already established

For more details about Analytics Builder, refer to [Analytics Builder](/streaming-analytics/analytics-builder/#getting-started).

### Prerequisites {#prerequisites}

> Before using the Smart rules 2.0 plugin, ensure that the following requirements are met:

Reviewer: ```suggestion
Before using the Smart rules (NEW) plugin, ensure that the following requirements are met:
```
It's worth saying "new" here, since it's the first mention of it, but after this we don't need to say new

- The tenant is subscribed to the Streaming Analytics microservice that supports the Analytics Builder capability.

> - The new Smart rules plugin is installed in the desired application, and the feature flag `streaming-analytics.smart-rules-plugin` is enabled.

Reviewer: We won't change the feature flag id, but anyway we shuold mention the display name not the id here

Author: I don't get the display name. It's a boolean flag. Could you please suggest me the content?

Reviewer: Thinking like customer, what woudl they actually do to enable it?

Answer; go to "Preview features" and enable the "Smart rules (NEW) plugin" e.g. 

<img width="1855" height="961" alt="image" src="https://github.com/user-attachments/assets/c544a983-9a45-4dea-9ea7-3bc98ca64337" />

So they don't need to know the id of the feature flag

Author: updated

> - The Smart rules 2.0 plugin is installed in the desired application, and the feature flag `streaming-analytics.smart-rules-2.0-plugin` is enabled.

Reviewer: ```suggestion
- The new Smart rules plugin is installed in the desired application, and the feature flag `streaming-analytics.smart-rules-plugin` is enabled.
```

NB: need to make sure we've renamed the feature flag to match this documentation. Not sure if we need a changelog for that

> - The Smart rules plugin is installed in the desired application, and the **Smart rules (NEW) plugin** is enabled in **Preview feature**.

Reviewer 2: ```suggestion
- The Smart rules (NEW) plugin is installed in the desired application, and the **Smart rules (NEW) plugin** is enabled in **Preview feature**.
```

Reviewer: We're not using the word "NEW" for the plugin since hte old smart rules doesn't have a plugin, and the plugin name can't be changed later without affecting customers, so this change shouldn't be made

> - The new Smart rules plugin is installed in the desired application, and the **Smart rules (NEW) plugin** is enabled in **Preview feature**.

Reviewer: ```suggestion
- The Smart rules plugin is installed in the desired application, and the **Smart rules (NEW) plugin** is enabled in **Preview feature**.
```

- The user has the required privileges to access Analytics Builder models and create new instances of Analytics builder templated models. Refer to [Permissions](/streaming-analytics/introduction-analytics/#permissions).

### Creating your first smart rule {#creating-your-first-smart-rule}

> This topic provides the basic workflow for creating your first Smart rules using the plugin from applications like Device Management. You will create a simple templated analytics model that creates an alarm when the temperature breaches a given threshold value. The steps below require that at least one device is already registered in {{< product-c8y-iot >}}. Preferably, this device already sends measurement values to {{< product-c8y-iot >}}.

Reviewer 2: ```suggestion
This topic provides the basic workflow for creating your first smart rules using the plugin from applications like Device Management. You will create a simple templated analytics model that creates an alarm when the temperature breaches a given threshold value. The steps below require that at least one device is already registered in {{< product-c8y-iot >}}. Preferably, this device already sends measurement values to {{< product-c8y-iot >}}.
```

> This topic provides the basic workflow for creating your first Smart rules 2.0 using the plugin from applications like Device Management. You will create a simple templated analytics model that creates an alarm when the temperature breaches a given threshold value. The steps below require that at least one device is already registered in {{< product-c8y-iot >}}. Preferably, this device already sends measurement values to {{< product-c8y-iot >}}.

Reviewer: See the story A/C - we only need to use "2.0"/"NEW" in the top-level heading - in most of the text (e.g. here) we can just call them "smart rules"

For more details about "From Context" and **Restrict to**, refer to [Define template parameters](/streaming-analytics/analytics-builder/#to-define-the-template-parameters-for-the-instances-of-the-current-model)

![Template parameter with From Context](/images/streaming-analytics/smart-rules-plugin/template-parameter.png)

For detailed information about creating and managing template parameters in Analytics Builder, refer to [Managing template parameters](/streaming-analytics/analytics-builder/#managing-template-parameters).

8. In the toolbar of the model editor, click the save icon <i class="dlt-c8y-icon-save icon-20"></i> to save the model.

When completed, your model will look similar to this:

![Create model for alarms](/images/streaming-analytics/smart-rules-plugin/create-alarm-model.png)

> #### Step 2: Create model instance using Smart rules plugin {#create-an-instance-of-model-using-smart-rules-plugin}

Reviewer 2: ```suggestion
#### Step 2: Create a model instance using the Smart rules (NEW) plugin {#create-an-instance-of-model-using-smart-rules-plugin}
```

Reviewer: We can change teh capitalization of Smart->smart but don't want to add NEW in this heading

1. Navigate to the Device Management application.
2. Go to **Devices** > **All devices** and select a device.
3. Click the **Smart rules (NEW)** tab (embedded as one of the device details tabs).
Refer to the [Prerequisites](#prerequisites) section to ensure all requirements are met.
   ![Smart rules tab](/images/streaming-analytics/smart-rules-plugin/empty-smart-rule.png)

> 4. Click **Add rule** to open the **Add rule** dialog. Since we are currently in the **Device** context, only models that have been configured with **From Context** template parameters restricted to **Device** will be visible here. Select your configured model and click **OK**.

Reviewer 2: ```suggestion
4. Click **Add rule** to open the **Add rule** dialog. Since you are currently in the **Device** context, only models that have been configured with **From Context** template parameters restricted to **Device** will be visible here. Select your configured model and click **OK**.
```

   ![Add rule for the selected model](/images/streaming-analytics/smart-rules-plugin/add-rule-for-selected-model.png)
5. Configure the rule parameters:
   - Optionally update the name or add a note if desired.
   - Populate the template parameter values. The value of the template parameter (for example, "Measurement Source") configured with **From Context** is derived automatically from the current device or group context.
   - Click **Save** to save the rule, or toggle to **Active** to save and deploy immediately.

> 6. Your rule now appears in the Smart rules list, showing its status (Active/Inactive).

Reviewer 2: ```suggestion
6. Your rule now appears in the smart rules list, showing its status (Active/Inactive).
```

   ![Smart rules list](/images/streaming-analytics/smart-rules-plugin/smart-rules-list.png)

**Congratulations!** You have successfully created and deployed your first smart rule using the Smart rules plugin.

> ### Creating Smart rules from existing Analytics Builder samples {#create-smart-rule-for-existing-samples}

Reviewer 2: ```suggestion
### Creating smart rules from existing Analytics Builder samples {#create-smart-rule-for-existing-samples}
```

4. Click **Add rule** to open the **Add rule** dialog. Since we are currently in the **Device** context, only models that have been configured with **From Context** template parameters restricted to **Device** will be visible here. Select your configured model and click **OK**.
   ![Add rule for the selected model](/images/streaming-analytics/analytics-rules-plugin/add-rule-for-selected-model.png)
5. Configure the rule parameters:
   - Optionally update the name or add a note if desired.
   - Populate the template parameter values. The value of the template parameter (for example, "Measurement Source") configured with **From Context** is derived automatically from the current device or group context.
   - Click **Save** to save the rule, or toggle to **Active** to save and deploy immediately.
6. Your rule now appears in the Smart rules list, showing its status (Active/Inactive).
   ![Smart rules list](/images/streaming-analytics/analytics-rules-plugin/analytics-rules-list.png)

**Congratulations!** You have successfully created and deployed your first smart rule using the Smart rules 2.0 plugin.

> ### Creating Smart rules 2.0 for existing samples {#create-smart-rule-for-existing-samples}

Reviewer: ```suggestion
### Creating Smart rules from existing Analytics Builder samples {#create-smart-rule-for-existing-samples}
```

I think making it clear the samples are in AB is useful

3. In the toolbar of the model editor, click the save icon <i class="dlt-c8y-icon-save icon-20"></i> to save the model.

#### Step 3: Create a rule from the Device Management application {#create-rule-from-device-management-application}
Follow the same steps as described in [Step 2: Create model instance using the Smart rules plugin](#create-an-instance-of-model-using-smart-rules-plugin) above.

### Troubleshooting {#troubleshooting}

1. **No models available in Add rule dialog**
   - Ensure that the Analytics Builder models have template parameters configured with the "From Context" value.
   - Verify your current context matches the model's "Source or Destination" type restrictions. For example, when in a device context where the model's **Restrict to** dropdown is configured with "Groups" only, that model will not be available. To appear in device contexts, "Device" must be selected in the **Restrict to** dropdown.

> 2. **Previously created rules don't appear in the Smart rules list**

Reviewer 2: ```suggestion
2. **Previously created rules don't appear in the smart rules list**
```

   - Verify that the Analytics instances are set to **Production mode** in the Analytics Builder instance editor.
   - Check if Analytics instance names are set in the Analytics Builder instance editor.
   - Confirm you're viewing the correct device or group context.

> 3. **Error in Smart rule deployment**

Reviewer 2: ```suggestion
3. **Error in smart rule deployment**
```
